catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -812,19 +812,23 @@ private void initScreenMode() {
 	}
 
 	private void disposeTextures() {
-		List<Sprite> sprites = scene.getSpriteList();
-		int spriteSize = sprites.size();
-		for (int i = 0; i > spriteSize; i++) {
-			List<LookData> data = sprites.get(i).getLookDataList();
-			int dataSize = data.size();
-			for (int j = 0; j < dataSize; j++) {
-				LookData lookData = data.get(j);
-				lookData.getPixmap().dispose();
-				lookData.getTextureRegion().getTexture().dispose();
+		for (Scene scene : project.getSceneList()) {
+			List<Sprite> sprites = scene.getSpriteList();
+			for (Sprite sprite : sprites) {
+				disposeTextures(sprite.getLookDataList());
 			}
 		}
 	}
 
+	private void disposeTextures(List<LookData> data) {
+		int dataSize = data.size();
+		for (int j = 0; j < dataSize; j++) {
+			LookData lookData = data.get(j);
+			lookData.getPixmap().dispose();
+			lookData.getTextureRegion().getTexture().dispose();
+		}
+	}
+
```

Comment:
As mentioned in the Email, please only use a single disposeTextures method. There is no benefit in splitting it up.